src/main.tsx

```diff
@@ -149,52 +157,93 @@ observer.observe(top.document.getElementById('main-content-container'), { attrib
+  });
+  const registerSetDiscourseContext = () =>
+    logseq.App.registerCommandPalette(
+      {
+        key: `discourse_setDiscourseContext`,
+        label: `Set Discourse Context`,
+        keybinding: {
+          mode: "global",
+          binding: "mod+shift+l",
+        },
       },
-    },
-    async () => {
-      const isEditing = await logseq.Editor.checkEditing();
-      if (!isEditing) {
-        logseq.UI.showMsg("Please edit (not just select) a block first.");
-        return;
+      async () => {
+        const isEditing = await logseq.Editor.checkEditing();
+        if (!isEditing) {
+          logseq.UI.showMsg("Please edit (not just select) a block first.");
+          return;
+        }
+        logseq.showMainUI();
+        await logseq.Editor.exitEditingMode(true);
+        const currBlock = await logseq.Editor.getCurrentBlock();
+        console.log("DB currBlock", currBlock);
+        const blockLocation = await logseq.Editor.queryElementRect(
+          `div[blockid="${currBlock?.uuid}"]`
+        );
+        const root = ReactDOM.createRoot(document.getElementById("app")!);
+        // console.log("DB blockLocation", blockLocation)
+        setTimeout(() => {
+          root.render(
+            <React.StrictMode>
+              {/* @ts-ignore */}
+              <Popup currBlockInfo={[currBlock, blockLocation]} />
+            </React.StrictMode>
+          );
+        }, 1000);
       }
-      logseq.showMainUI();
-      await logseq.Editor.exitEditingMode(true)
-      const currBlock = await logseq.Editor.getCurrentBlock();
-      console.log("DB currBlock", currBlock)
-      const blockLocation = await logseq.Editor.queryElementRect(`div[blockid="${currBlock?.uuid}"]`)
-      const root = ReactDOM.createRoot(document.getElementById("app")!);
-      // console.log("DB blockLocation", blockLocation)
-      setTimeout(
-        ()=>{root.render(
-          <React.StrictMode>
-            {/* @ts-ignore */}
-          <Popup currBlockInfo={[currBlock, blockLocation]} />
-          </React.StrictMode>
-        );}, 1000
-      )
-          
-    });
-    registerSetDiscourseContext()
-
-  const registerupdateDiscourceCSS = () => logseq.App.registerCommandPalette(
-    {
-      key: `discoursse_updateDiscourceCSS`,
-      label: `Update Discource CSS`,
-      keybinding: {
-        mode: 'global',
-        binding: 'mod+shift+;'
+    );
+  registerSetDiscourseContext();
+
+  const registerupdateDiscourceCSS = () =>
+    logseq.App.registerCommandPalette(
+      {
+        key: `discoursse_updateDiscourceCSS`,
+        label: `Update Discource CSS`,
+        keybinding: {
+          mode: "global",
+          binding: "mod+shift+;",
+        },
       },
-    },
-    async () => { updateDiscourceCSS() });
-  registerupdateDiscourceCSS()
+      async () => {
+        updateDiscourceCSS();
+      }
+    );
+  registerupdateDiscourceCSS();
+  console.log("get api");
+  const samepageUnloader = await getSamePageAPI().then(
```

Comment:
this is the only actual set of changes.

Need to keep the function async and add the unload callback instead of awaiting and hanging up the rest of the extension